plotPattern.py

```diff
@@ -0,0 +1,544 @@
+
+Plot the pattern for New Chip Test Box (.pat)
+
+Changes:
+    - 2017-11-21 Adapt it to python-3
+    - 2017-09-25 All can be plotted
+    - 2017-09-22 Can be plotted but the loop and wait not work yet
+
+@author: zhang_j1
+"""
+import matplotlib.pyplot as plt
+from numpy import *
+from matplotlib.pyplot import *
+from matplotlib.patches import Rectangle
+import os
+import argparse
+
+class PlotPattern():
+    def __init__(self, pattern, signalNames, colors_plot, colors_wait, linestyles_wait, alpha_wait, alpha_wait_rect, colors_loop, linestyles_loop, alpha_loop, alpha_loop_rect, clock_vertical_lines_spacing, show_clocks_number, line_width):
+        
+        self.pattern = pattern
+        self.signalNames = signalNames
+        self.verbose = False
+        # TODO: send alias
+
+        self.colors_plot = colors_plot.copy()
+        self.colors_wait = colors_wait.copy()
+        self.linestyles_wait = linestyles_wait.copy()
+        self.alpha_wait = alpha_wait.copy()
+        self.alpha_wait_rect = alpha_wait_rect.copy()
+        self.colors_loop = colors_loop.copy()
+        self.linestyles_loop = linestyles_loop.copy()
+        self.alpha_loop = alpha_loop.copy()
+        self.alpha_loop_rect = alpha_loop_rect.copy()
+        self.clock_vertical_lines_spacing = clock_vertical_lines_spacing
+        self.show_clocks_number = show_clocks_number     
+        self.line_width = line_width     
+
+        self.colors_plot[0] = f'xkcd:{colors_plot[0].lower()}'
+        self.colors_plot[1] = f'xkcd:{colors_plot[1].lower()}'
+        
+        for i in range(6):
+            self.colors_wait[i] = f'xkcd:{colors_wait[i].lower()}'
+            self.colors_loop[i] = f'xkcd:{colors_loop[i].lower()}'
+
+        if self.verbose:
+            self.printPatViewerParameters()
+
+    def printPatViewerParameters(self):
+        print('Pattern Viewer Parameters:')
+        print(f'\tcolor1: {self.colors_plot[0]}, color2: {self.colors_plot[1]}')
+        print(f"\twait color: {self.colors_wait}")
+        print(f"\twait linestyles: {self.linestyles_wait}")
+        print(f"\twait alpha: {self.alpha_wait}")
+        print(f"\twait alpha rect: {self.alpha_wait_rect}")
+        print(f"\tloop color: {self.colors_loop}")
+        print(f"\tloop linestyles: {self.linestyles_loop}")
+        print(f"\tloop alpha: {self.alpha_loop}")
+        print(f"\tloop alpha rect: {self.alpha_loop_rect}")
+        print(f'\tclock vertical lines spacing: {self.clock_vertical_lines_spacing}')
+        print(f'\tshow clocks number: {self.show_clocks_number}')
+        print(f'\tline width: {self.line_width}')
+        print('\n')
+
+
+    def dec2binary(self, dec_num, width=None):
+        return binary_repr(int(dec_num), width=width)
+
+    def hex2dec(self, string_num):
+        return str(int(string_num.upper(), 16))
+
+    def hex2binary(self, string_num, width=None):
+        return self.dec2binary(self.hex2dec(string_num.upper()), width=width)
+
+    def patternPlot(self):
```

Comment:
It looks like the function is doing loading, decoding and plotting. For testing it would be great if we could split it up in different functions...

Since we need to move forward we can keep it now but revisit in the future?

Reply:
This whole file is written by Jiaguo and Julian. I am not even sure how it does the pattern plotting. But yes, we could refactor or try to understand it in the future